### ISSUE TITLE
Sentinel executes templates for all matching watchers regardless of prefix

### ISSUE THREAD

**Issue by mdkent** [BlueDragonX/sentinel#8]
Hi! Thanks for writing sentinel, it's proved very useful.

In doing some further testing I noticed a fun behaviour: when updating a watched key, sentinel will execute templates for any other watchers observing the same key, regardless of prefix. eg: Given the config:

```yaml
etcd:
  uri: http://localhost:4001/
  prefix: irrelevant
logging:
  level: debug
watchers:
  config_one:
    prefix: one
    watch:
      - foo/_trigger
    context:
      - foo/stuff
    templates:
      - src: /etc/sentinel.d/test/generic.tpl
        dest: /tmp/one.yml

  config_two:
    prefix: two
    watch:
      - foo/_trigger
    context:
      - foo/stuff
    templates:
      - src: /etc/sentinel.d/test/generic.tpl
        dest: /tmp/two.yml

  config_three:
    prefix: three
    watch:
      - foo/_renamed_trigger
    context:
      - foo/stuff
    templates:
      - src: /etc/sentinel.d/test/generic.tpl
        dest: /tmp/three.yml
```
and running
```
 curl -s -L -XPUT -d value="test" http://localhost:4001/v2/keys/one/foo/_trigger
```
You'd only expect it to update the template for the 'config_one' watcher, based on the prefix, but instead config_one and config_two are run when the foo/_trigger names are matched:

```
sentinel 2015/01/15 23:21:41.067530 [DEBUG]  key '/one/foo/_trigger' changed
sentinel 2015/01/15 23:21:41.068889 [DEBUG]  get key '/two/foo/stuff': &{/two/foo/stuff  true <nil> 0 [0xc20807bda0] 82962 82962}
sentinel 2015/01/15 23:21:41.068908 [DEBUG]  context: map[foo:map[stuff:map[important:test]]]

sentinel 2015/01/15 23:21:41.069071 [DEBUG]  template '/tmp/two.yml' did not change
sentinel 2015/01/15 23:21:41.069076 [INFO]   config_two skipped execution
sentinel 2015/01/15 23:21:41.069526 [DEBUG]  get key '/one/foo/stuff': &{/one/foo/stuff  true <nil> 0 [0xc2080f8720] 82960 82960}
sentinel 2015/01/15 23:21:41.069543 [DEBUG]  context: map[foo:map[stuff:map[important:test]]]

sentinel 2015/01/15 23:21:41.069646 [DEBUG]  template '/u/one.yml' did not change
sentinel 2015/01/15 23:21:41.069651 [INFO]   config_one skipped execution
```
config_three, with it's different watch key, is spared.

I've also confirmed commands will be executed as well, which might surprise someone.

Suggesting they be tied to the prefix as is suggested in the README, or adding a warning about using similar watch keys.

**Comment by BlueDragonX** [comment]
Thanks for the discovery! I'll include this in the round of fixes I'm working on. I'm hoping to have a new release sometimes next week.

**Comment by BlueDragonX** [comment]
Fixed in v2.0.0.
